esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -2884,8 +2884,10 @@ INewResultSet* createFilteredResultSet(INewResultSet* result, IArrayOf<IConstNam
 
 static bool isResultRequestSzTooBig(unsigned __int64 start, unsigned requestCount, unsigned __int64 resultSz, unsigned resultRows, unsigned __int64 limitSz)
 {
-    if ((0 == requestCount) || (0 == resultRows))
+    if ((0 == start) && (0 == requestCount)) // all being requested
         return resultSz > limitSz;
+    else if (0 == resultRows) // assume unknown number of rows
```

Comment:
Perhaps "Insufficient information available to estimate average row size" would be a better comment